viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 
+def add_to_watched(user_data, movie):
+    user_data["watched"]= [movie]
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"]= [movie]
+    return user_data
+
+def watch_movie(user_data, movie):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == movie:
+            user_data["watched"].append(movie_dict)
+            user_data["watchlist"].remove(movie_dict)   
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_list = [movie_dict["rating"]for movie_dict in user_data["watched"]]
+    if rating_list != []:
+        average = sum(rating_list)/len(user_data["watched"])
+    else:
+        average = 0.0
+    return average
 
+def get_most_watched_genre(user_data):
+    genre_list = [movie_dict["genre"] for movie_dict in user_data["watched"]]
+
+    if genre_list != []:
+        popular_genre = statistics.mode(genre_list)
+        return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+# ----------------------------------------
+
+def get_unique_watched(user_data):
+    # lists of movies watched by user, and by friends 
+    movies_user_watched = []
+    movies_friends_watched = []
+
+    # add movie titles to the lists
+    for i in range(len(user_data["watched"])):
+        movie_title = user_data["watched"][i]['title']
+        movies_user_watched.append(movie_title)
+
+    for i in range(len(user_data["friends"])):
+        list_of_dicts = user_data["friends"][i]['watched']
+        for i in range(len(list_of_dicts)):
+            movies_friends_watched.append(list_of_dicts[i]['title'])
+    
+    # use set to compare the lists, taking the difference between the user's set and the friends set.
+    unique_user_set = set(movies_user_watched) - set(movies_friends_watched)
+
+    # convert set of movie titles --> list of dictionaries of those movies
+    unique_list_of_dicts = []
+    for movie in unique_user_set:
+        for i in range(len(user_data["watched"])):
+            if movie == user_data["watched"][i]['title']:
+                unique_list_of_dicts.append(user_data["watched"][i])
+
+    return unique_list_of_dicts
+
+def get_friends_unique_watched(user_data):
+    # lists of movies watched by user, and by friends 
+    movies_user_watched = []
+    movies_friends_watched = []
+
+    # add movie titles to the lists
+    for i in range(len(user_data["watched"])):
+        movie_title = user_data["watched"][i]['title']
+        movies_user_watched.append(movie_title)
+
+    for i in range(len(user_data["friends"])):
+        list_of_dicts = user_data["friends"][i]['watched']
+        for i in range(len(list_of_dicts)):
+            movies_friends_watched.append(list_of_dicts[i]['title'])

```

Comment:
This section is identical to lines 60-67.  I recommend pulling this section out into a helper function(s) to DRY (Don't Repeat Yourself) your code to improve readability.